### ISSUE TITLE
Feature request: Make notes on HTML files

### ISSUE THREAD

**Issue by 4t0m** [weirdNox/org-noter#82]
I'm not sure how hard this would be to implement but I might be interested in working on it.  Do you have any pointers on how I would go about doing this?

**Comment by kbruskiewicz** [comment]
I think the package author's philosophy is that he doesn't deal with data that can't paginate.

That said EPUB files are "just" zipped up HTML files, so you could ostensibly have a workflow that crosses over to that format. Ditto rendering HTML in Word files.

I agree with the use case though.

**Comment by kbruskiewicz** [comment]
You could also print the buffer to postscript with `ps-print-buffer`, which _should_ be DocView compatible.

**Comment by weirdNox** [comment]
Yeah, things that don't paginate or that are not stable (eg. a PDF that is modified, and things move around) don't work too well with org-noter.

As for EPUBs, I do it based on the `(point)` selected by the user on the buffer rendered by nov.el, so it stops working if `shr` starts rendering it differently (in a way that modifies the character count).

If the HTML is static, like a page you download for example, it could be done if you really want it.
You just need a way to obtain a value or reference given by the user that can non-ambiguously be dereferenced. Even a string search may work (like `org-noter-pdftools`).

This summer, when `org-noter-pdftools` was proposed, instead of merging it in, I started abstracting functionality into several functions that may have augmented functionality via hooks. If you see [the code I made for `org-noter-pdftools`](https://github.com/weirdNox/org-noter/blob/master/other/org-noter-integration.el), you'll see that it has functions (eg. `org-noter-pdftools--convert-to-location-cons`) that are put in hooks (eg. `org-noter--convert-to-location-cons-hook`) that are then called by `org-noter` when applicable. 

The list of hooks available is on the top of `org-noter.el`, and more can be added if you need them.

I hope this gave you some insight on how to do what you want :)

**Comment by bepolymathe** [comment]
Hi,

This discussion makes me think of one thing. Org-noter is an essential tool in my daily work as a researcher. I am also curious about the development of [Polar](https://getpolarized.io/), which seeks to make it possible to annotate pdf and html for a wider audience. 

The solution he proposes for [html](https://getpolarized.io/docs/web-page-capture-and-archival-of-html-content.html) is a static format... but I don't know if docview or Emacs in general could read it one day...
